wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarSendController.swift

```diff
@@ -29,63 +29,77 @@ final class ConversationInputBarSendController: NSObject {
         super.init()
     }
 
-    func sendMessage(withImageData imageData: Data,
-                     userSession: UserSession,
-                     completion completionHandler: Completion? = nil) {
+    func sendMessage(
+        withImageData imageData: Data,
+        userSession: UserSession,
+        completion completionHandler: Completion? = nil
+    ) {
 
         guard let conversation = conversation as? ZMConversation else { return }
 
         feedbackGenerator.prepare()
         userSession.enqueue({
             do {
-                try conversation.appendImage(from: imageData)
+                let useCase = userSession.makeAppendImageMessageUseCase()
+                try useCase.invoke(withImageData: imageData, in: conversation)
                 self.feedbackGenerator.impactOccurred()
             } catch {
                 Logging.messageProcessing.warn("Failed to append image message. Reason: \(error.localizedDescription)")
             }
-        }, completionHandler: {
-            completionHandler?()
```

Comment:
You should still call the completion handler.

Reply:
Done: https://github.com/wireapp/wire-ios/commit/fa7fd0216a50d284439dd44f6136314f57d44b31